optimizely/client/factory.go

```diff
@@ -25,18 +25,21 @@ import (
 	"github.com/optimizely/go-sdk/optimizely/decision"
 )
 
+const datafileURLTemplate = "https://cdn.optimizely.com/datafiles/%s.json"
+
 // OptimizelyFactory is used to construct an instance of the OptimizelyClient
 type OptimizelyFactory struct {
-	SDKKey   string
-	Datafile []byte
+	SDKKey               string
+	Datafile             []byte
+	ProjectConfigManager optimizely.ProjectConfigManager
```

Comment:
In Java the factory has only static methods, and the only component shared among instances it creates is the logger. Are we sure we want to do this instead (associating a project config manager with the factory and applying it to all instances created in factory methods)?

Reply:
Hmm good point. I will go get some inspiration and refactor this.

Reply:
I like the changes, but is `OptimizelyFactory` struct still supposed to have this `ProjectConfigManager` field?